scripts/eurostat/health_determinants/bmi/process.py

```diff
@@ -0,0 +1,555 @@
+                 "typeOf: dcs:StatisticalVariable\n"
+                 "populationType: dcs:Person\n"
+                 "statType: dcs:measuredValue\n"
+                 "measuredProperty: dcs:count\n"
+                 "{xtra_pvs}\n")
+
+_TMCF_TEMPLATE = (
+    "Node: E:eurostat_population_bmi->E0\n"
+    "typeOf: dcs:StatVarObservation\n"
+    "variableMeasured: C:eurostat_population_bmi->SV\n"
+    "measurementMethod: C:eurostat_population_bmi->Measurement_Method\n"
+    "observationAbout: C:eurostat_population_bmi->geo\n"
+    "observationDate: C:eurostat_population_bmi->time\n"
+    "value: C:eurostat_population_bmi->observation\n")
+
+_INCOME_QUINTILE_VALUES = {
+    "IncomeOf0To20Percentile": "[0 20 Percentile]",
+    "IncomeOf20To40Percentile": "[20 40 Percentile]",
+    "IncomeOf40To60Percentile": "[40 60 Percentile]",
+    "IncomeOf60To80Percentile": "[60 80 Percentile]",
+    "IncomeOf80To100Percentile": "[80 100 Percentile]",
+}
+
+_EXISTING_SV_DEG_URB_GENDER = {
+    "Count_Person_Female_Rural": "Count_Person_Rural_Female",
+    "Count_Person_Male_Rural": "Count_Person_Rural_Male",
+    "Count_Person_Female_Urban": "Count_Person_Urban_Female",
+    "Count_Person_Male_Urban": "Count_Person_Urban_Male",
+}
+
+_SV_DENOMINATOR = 1
+
+
+def _age_sex_education(data_df: pd.DataFrame) -> pd.DataFrame:
+    """
+    Returns the Cleaned DataFrame consists
+    education with bmi,age,and sex combination data for the year 2014, 2019.
+    Args:
+        data_df (pd.DataFrame): DataFrame having raw data
+
+    Returns:
+        pd.DataFrame: Cleaned DataFrame
+    """
+    df_cols = ['unit,bmi,isced11,sex,age,geo', '2019', '2014']
+    data_df.columns = df_cols
+    multiple_cols = "unit,bmi,isced11,sex,age,geo"
+    # Splits the DataFrame having mulitple column data
+    # in a single column and creates them as Individual Columns
+    data_df = _split_column(data_df, multiple_cols)
+    # Filtering out the required rows and columns
+    data_df = data_df[data_df['age'] == 'TOTAL']
+    data_df = data_df[~(data_df['geo'].isin(['EU27_2020', 'EU28']))]
+    data_df = _replace_bmi(data_df)
+    data_df = _replace_sex(data_df)
+    data_df = _replace_isced11(data_df)
+    # Creating SV using the DataFrame Values
+    data_df['SV'] = 'Percent_' + data_df['bmi'] + '_' + \
+                    'In_Count_Person_' + data_df['isced11'] +\
+                    '_' + data_df['sex']
+    data_df = data_df.drop(columns=['unit', 'age', 'isced11', 'bmi', 'sex'])
+    data_df = data_df.melt(id_vars=['SV','geo'], var_name='time'\
+            ,value_name='observation')
+    return data_df
+
+
+def _age_sex_education_history(data_df: pd.DataFrame) -> pd.DataFrame:
+    """
+    Returns the Cleaned DataFrame consists
+    education with bmi,age,and sex combination data for the year 2008.
+    Args:
+        data_df (pd.DataFrame): DataFrame having raw data
+
+    Returns:
+        pd.DataFrame: Cleaned DataFrame
+    """
+    df_cols = [
```

Comment:
curious why we're defining the column labels here, do the column labels not come in the file?

Reply:
We will get column labels from dataset, initially we observed couple of issues like labels having names such as 'geo\time', 'time\geo' and extra spaces in labels. Now added code to handle these scenarios like stripping spaces at initial stages and maintaining a dictionary to rename the labels.